code/modules/mob/living/silicon/ai/robot_control.dm

```diff
@@ -58,23 +72,47 @@
 	if(!is_interactable(usr))
 		return
 
-	var/mob/living/simple_animal/bot/bot
+	var/mob/living/simple_animal/bot/bot = locate(params["ref"]) in GLOB.bots_list
+	var/mob/living/basic/bot/basic_bot = locate(params["ref"]) in GLOB.basic_bots_list
+
+	if(owner.control_disabled)
+		return
+
 	switch(action)
 		if("callbot") //Command a bot to move to a selected location.
 			if(owner.call_bot_cooldown > world.time)
 				to_chat(usr, span_danger("Error: Your last call bot command is still processing, please wait for the bot to finish calculating a route."))
 				return
-			bot = locate(params["ref"]) in GLOB.bots_list
-			owner.bot_ref = WEAKREF(bot)
-			if(!bot || !(bot.bot_mode_flags & BOT_MODE_REMOTE_ENABLED) || owner.control_disabled)
+
+			if(bot)
+				owner.bot_ref = WEAKREF(bot)
+
+				if(bot.bot_mode_flags & BOT_MODE_REMOTE_ENABLED)
+					owner.waypoint_mode = TRUE
+					to_chat(usr, span_notice("Set your waypoint by clicking on a valid location free of obstructions."))
+					. = TRUE
```

Comment:
switch all these `.` uses in this proc to just return. Should be fine since actions are a switch case, and we're trying to minimize unnecessary use